Sources/SwiftSyntax/SyntaxConvenienceMethods.swift

```diff
@@ -0,0 +1,81 @@
+//===- SyntaxConvenienceMethods.swift - Convenience funcs for syntax nodes ===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2020 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+public extension FloatLiteralExprSyntax {
+  var floatingValue: Double {
+    let floatingDigitsWithoutUnderscores = floatingDigits.text.filter {
+      $0 != "_"
+    }
+    return Double(floatingDigitsWithoutUnderscores)!
+  }
+}
+
+public extension IntegerLiteralExprSyntax {
+  var integerValue: Int {
+    let text = self.digits.text
+    let type = IntegerType(text: text)
+    let textWithoutPrefix: String
+    let digitsStartIndex = text.index(text.startIndex, offsetBy: type.prefixLength())
+    textWithoutPrefix = String(text.suffix(from: digitsStartIndex))
+
+    let textWithoutPrefixOrUnderscores = textWithoutPrefix.filter {
+      $0 != "_"
+    }
+
+    return Int(textWithoutPrefixOrUnderscores, radix: type.radix())!
+  }
+
+  private enum IntegerType {
```

Comment:
I feel like this type is overkill. It could be a single inner function returning a tuple of `prefixLength` and `radix`

Reply:
Done.